### ISSUE TITLE
[Android]: App shows white screen after reopening the app in emulator running api 35 in release mode. 

### ISSUE THREAD

**Issue by asemanTest** [flutter/flutter#151027]
### Steps to reproduce

 in android 15(Sdk 35) When running the application with Flutter version 3.22.2, it works correctly on the first run. However, after closing it and sending it to the background, reopening it displays a white screen.
i also create new flutter app with default setting and code in release and debug mode but it's not work again.

### Expected results

show app screen correctly

### Actual results

white screen

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

![Screenshot_1719642078](https://github.com/flutter/flutter/assets/90551408/9e56e8ad-ac26-4a34-acc1-43642cf3cb94)

</details>

### Logs

<details open><summary>Logs</summary>

```console
Launching lib\main_dev.dart on sdk gphone64 x86 64 in debug mode...
Running Gradle task 'assembleDebug'...
√ Built build\app\outputs\flutter-apk\app-debug.apk
D/FlutterGeolocator( 8337): Attaching Geolocator to activity
D/FlutterGeolocator( 8337): Creating service.
D/FlutterGeolocator( 8337): Binding to location service.
D/FlutterLocationService( 8337): Creating service.
D/FlutterLocationService( 8337): Binding to location service.
D/FlutterGeolocator( 8337): Geolocator foreground service connected
D/FlutterGeolocator( 8337): Initializing Geolocator services
D/FlutterGeolocator( 8337): Flutter engine connected. Connected engine count 1
I/Choreographer( 8337): Skipped 53 frames!  The application may be doing too much work on its main thread.
Debug service listening on ws://127.0.0.1:52493/1E-h9GuqOlI=/ws
Syncing files to device sdk gphone64 x86 64...
W/( 8337): Verification of boolean com.google.android.gms.common.zzn.zzf() took 223.474ms (156.62 bytecodes/s) (3456B approximate peak alloc)
I/( 8337): Compiler allocated 4968KB to compile void android.view.ViewRootImpl.performTraversals()
D/ProfileInstaller( 8337): Installing profile for com.example.test
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.22.2, on Microsoft Windows [Version 10.0.19045.4412], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2019 16.11.19)
[√] Android Studio (version 2023.3)
[√] VS Code (version 1.69.2)
[√] Connected device (5 available)
[√] Network resources

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
@asemanTest 
I tried this by creating android 15 emulator and ran a sample code on it with latest stable. Sent the app to background and opened it again which displayed splash screen for few seconds and then the actual screen. I didn't observe the blank screen after opening it from background. 

Are you seeing this in physical device or emulator ?

**Comment by asemanTest** [comment]
> @asemanTest I tried this by creating android 15 emulator and ran a sample code on it with latest stable. Sent the app to background and opened it again which displayed splash screen for few seconds and then the actual screen. I didn't observe the blank screen after opening it from background.
> 
> Are you seeing this in physical device or emulator ?

i use android studio emulator (api level 35 ,Google play intel x86_64 atom system image)
i created a new app and test again you can see the result below:
![untitled](https://github.com/flutter/flutter/assets/90551408/8d27d2dc-6786-4498-88b1-4f4b07d7ef01)

**Comment by asemanTest** [comment]
@darshankawar 
The app only runs again if the cache is cleared; otherwise, the issue persists with hot reload and hot restart.

**Comment by darshankawar** [comment]
Thanks for the update @asemanTest 
I again tried by creating new Flutter app but didn't observe the behavior. See video below:


https://github.com/flutter/flutter/assets/67046386/e8d94369-a579-41d7-9dfb-48475494b351

**Comment by darshankawar** [comment]
@asemanTest 
I observed that if you are using any plugin, let's says go_router, and then run in release mode, it seems to stuck at splash screen as below:


https://github.com/flutter/flutter/assets/67046386/1d329f68-3ff6-4c16-a51b-243ce44aee5d

But the same works properly in api 33 as below:


https://github.com/flutter/flutter/assets/67046386/1c8557bf-dfd7-4ea6-9d55-35f43a6b80cb

But if you create a new Flutter app (counter app) and run in release mode, it works as expected. 



> in release mode, this issue still exists

Is this with counter app or using any plugin ?

**Comment by asemanTest** [comment]
I use the auto_route package for routing in my apps.
In apps that use this package, and in the counter app, this issue exists in release mode.

**Comment by darshankawar** [comment]
> n apps that use this package, and in the counter app, this issue exists in release mode.

And this is only occuring while using api 35 and not with < api 35 ?

**Comment by asemanTest** [comment]
> > n apps that use this package, and in the counter app, this issue exists in release mode.
> 
> And this is only occuring while using api 35 and not with < api 35 ? Also, can you check if using api 35 with 3.19.6 version, do you still get same stuck behavior or not ?

I checked the counter app in Flutter 3.19.6 on API 35 and the issue exists again. I guess this is an emulator bug because everything is okay on my phone with API 34

**Comment by darshankawar** [comment]
I am unable to replicate per my earlier verification with counter app running on api 35 emulator in release mode. Based on the report, I will keep the issue open for team's attention.

**Comment by timbotimbo** [comment]
@asemanTest in you screen capture with the counter app above, can you still press the + button even though it is not visible?
I ran into an issue that looks very similar, but on API versions lower than 35. 
And in my case you still see the ink or shadow effect of the button, even though the rest of the app is blank.

**Comment by kariyum** [comment]
I'm also experiencing this exact issue. This is what I get when I run the counter app for the second time.
![image](https://github.com/flutter/flutter/assets/78426214/461d8bff-dc4e-4af6-b9d7-5ac546a2aefd)

Note that it runs perfectly fine for the first install but after that it doesn't load fully. Clearing the app cache in the emulator removes the issue. 
Flutter version 3.22.2 on channel stable
Platform android-35, build-tools 35.0.0
Java version OpenJDK Runtime Environment (build 17.0.10+0--11609105)
sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 15 (API 35) (emulator)

**Comment by kariyum** [comment]
UPDATE:
Following https://stackoverflow.com/questions/48840825/error-0x502-android-emulator, chanding the "Emulated Performance: Graphics to Software" solves this issue. So this may be related to my GPU nvidia driver

Cheers,

**Comment by aumbhusal** [comment]
![Screenshot 2024-07-09 204150](https://github.com/flutter/flutter/assets/139211028/52461b9c-c6a1-4bc6-9407-13031a58f4bb)
Same error

**Comment by idhpaul** [comment]
I got the same issue too.
While looking through the GitHub issue lists, I found a common factor.
Everyone is using Flutter version 3.22.2.

Currently, I am developing on a Galaxy Book 4 Ultra. 
It has both Intel GPU(Intl Arc Graphics /driver version : 31.0.101.5333) and NVIDIA GPU(RXT 4050 Laptop GPU / driver version : 556.12)
Could this environment be causing the issue?

**Comment by zhoulijun12315** [comment]
i got the same issue too.

Only in release mode, will show white screen and App will be stop in this screen.But in debug mode, everything is ok.

`Flutter 3.22.2 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 761747bfc5 (8 周前) • 2024-06-05 22:15:13 +0200
Engine • revision edd8546116
Tools • Dart 3.4.3 • DevTools 2.34.3`

**Comment by Spitzbua** [comment]
Same behaviour on my side. On first app install everything is working fine. Stopping the process and restarting (Launching lib\main.dart on sdk gphone64 x86 64 in debug mode.) be stuck on the splash screen. 
Brandnew workstation i9 / RTX 3500

**Comment by rayliverified** [comment]
For anyone still running into this issue, this is a known Android Emulator issue that affects Windows 10 and 11 devices.
The recommended fix is to remove all Android Emulators, SDKs, etc from the SDK Manager and reinstall the latest versions.

Unfortunately, the latest version has the same issue so it is not a true fix.

As of now, the Android emulator image is broken on Windows.

**Comment by gmackall** [comment]
It seems likely to me that this is another instance of https://issuetracker.google.com/issues/345674553.

Can you try on emulator version `35.1.18` (following the instructions at https://developer.android.com/studio/emulator_archive), and see if that fixes your issue?

**Comment by rd-wang** [comment]
save issues in the latest  version
os windows11
emulator  34.2.16  API 35  x86_64
Flutter (Channel stable, 3.24.0, on Microsoft Windows [版本 10.0.22631.4037], locale zh-CN)

![image](https://github.com/user-attachments/assets/a5285ca6-9aa0-44b9-992b-da84c3d9d2e3)

**Comment by gmackall** [comment]
@rd-wang 
The emulator bug is only marked as fixed on versions 35.1.18 or 35.2.4 or higher, can you try with either of those versions?
